src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ISafeProjectGuidService.cs

```diff
@@ -1,21 +1,18 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
 
 using Microsoft.VisualStudio.Composition;
-using Microsoft.VisualStudio.ProjectSystem.VS;
-
-#pragma warning disable RS0030 // This is the one place where IProjectGuidService is allowed to be referenced
 
 namespace Microsoft.VisualStudio.ProjectSystem
 {
     /// <summary>
-    ///     Provides a mechanism to safely access the project GUID. Replaces usage of <see cref="IProjectGuidService"/>
-    ///     and <see cref="IProjectGuidService2"/>.
+    ///     Provides a mechanism to safely access the project GUID. Replaces usage of the IProjectGuidService
+    ///     and IProjectGuidService2 interfaces from CPS which are not safe to use in all contexts.
     /// </summary>
     /// <remarks>
-    ///     <see cref="IProjectGuidService"/> and <see cref="IProjectGuidService2"/> will retrieve the project GUID of
-    ///     the project *at the time* that it is called. During project initialization, the GUID may be changed by the
-    ///     solution in reaction to a clash with another project. <see cref="ISafeProjectGuidService"/> will wait until
-    ///     it is safe to retrieve the project GUID before returning it.
+    ///     IProjectGuidService and IProjectGuidService2 will retrieve the project GUID of the project *at the time*
+    ///     that it is called. During project initialization, the GUID may be changed by the solution in reaction to a
+    ///     clash with another project. <see cref="ISafeProjectGuidService"/> will wait until it is safe to retrieve
+    ///     the project GUID before returning it.
```

Comment:
Since `IProjectGuidService` and `IProjectGuidService2` are VS layer specific, shouldn't these comments exist in where `ISafeProjectGuidService` is used and not on this interface?

Reply:
I think this is a better place to explain **why** `ISafeProjectGuidService` exists. But the issues with layering do make this a little funky, yes.